internal/collector/collector_test.go

```diff
@@ -230,6 +231,229 @@ var _ = Describe("Factory#NewPodSpec", func() {
 				verifyProxyVolumesAndVolumeMounts(collector, podSpec, "custom-clf-trustbundle")
 			})
 		})
+
+		Context("and mounting volumes", func() {
+			It("should mount host path volumes", func() {
+				Expect(podSpec.Volumes).To(HaveLen(15))
+				Expect(podSpec.Volumes).To(ContainElement(v1.Volume{Name: logContainers, VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: logContainersValue}}}))
+			})
+		})
+	})
+
+})
+
+var _ = Describe("Factory#Deployment#NewPodSpec", func() {
+	var (
+		podSpec   v1.PodSpec
+		collector v1.Container
+
+		factory *Factory
+	)
+	BeforeEach(func() {
+		factory = &Factory{
+			CollectorType: logging.LogCollectionTypeFluentd,
+			ImageName:     constants.FluentdName,
+			Visit:         fluentd.CollectorVisitor,
+			ResourceNames: coreFactory.GenerateResourceNames(*runtime.NewClusterLogForwarder(constants.OpenshiftNS, constants.SingletonName)),
+			isDaemonset:   false,
+		}
+		podSpec = *factory.NewPodSpec(nil, logging.ClusterLogForwarderSpec{}, "1234", "", tls.GetClusterTLSProfileSpec(nil), nil, constants.OpenshiftNS)
+		collector = podSpec.Containers[0]
+	})
+	Describe("when creating of the collector container", func() {
+
+		It("should provide the pod IP as an environment var", func() {
+			Expect(collector.Env).To(IncludeEnvVar(v1.EnvVar{Name: "POD_IP",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						APIVersion: "v1", FieldPath: "status.podIP"}}}))
+		})
+		It("should set a security context", func() {
+			Expect(collector.SecurityContext).To(Equal(&v1.SecurityContext{
```

Comment:
We could also remove this context I'm betting all-to-gether.  This defines the cgroups for the containers which we do not need anything special in this scenario